client/telemetry/TelemetryEventClient.go

```diff
@@ -105,9 +106,24 @@ func NewTelemetryEventClientImpl(logger *zap.SugaredLogger, client *http.Client,
 		logger.Errorw("error in starting heartbeat event", "err", err)
 		return nil, err
 	}
+	err = watcher.UpdateCloudProviderCache()
+	if err != nil {
+		logger.Errorw("error in updating cloud provider", "err", err)
+		return nil, err
+	}
```

Comment:
this will slow startup, instead lets do this for all calls of getting identifyProvider where we first check if this has been set and if not then call the process